File: resources/capability-bundle/.claude/skills/post-event-review/scripts/align-turns.mjs
const payload = JSON.parse(process.argv[2] ?? '{}');
const turns = Array.isArray(payload.turns) ? payload.turns : [];

const normalized = turns.map((turn, index) => ({
  ref: String(turn.ref ?? `turn-${index + 1}`),
  speaker: String(turn.speaker ?? 'unknown'),
  start_ms: Number(turn.start_ms ?? 0),
  end_ms: Number(turn.end_ms ?? turn.start_ms ?? 0),
  text: String(turn.text ?? '').trim(),
})).filter((turn) => turn.text);

normalized.sort((left, right) => left.start_ms - right.start_ms);
process.stdout.write(JSON.stringify({ turns: normalized }));

File: resources/capability-bundle/.claude/skills/post-event-review/scripts/align-turns.py
import json
import sys


payload = json.loads(sys.argv[1] if len(sys.argv) > 1 else "{}")
turns = payload.get("turns") if isinstance(payload.get("turns"), list) else []

normalized = []
for index, turn in enumerate(turns):
    turn = turn if isinstance(turn, dict) else {}
    text = str(turn.get("text", "")).strip()
    if not text:
        continue
    start = float(turn.get("start_ms", 0))
    normalized.append({
        "ref": str(turn.get("ref", f"turn-{index + 1}")),
        "speaker": str(turn.get("speaker", "unknown")),
        "start_ms": start,
        "end_ms": float(turn.get("end_ms", start)),
        "text": text,
    })

normalized.sort(key=lambda turn: turn["start_ms"])
sys.stdout.write(json.dumps({"turns": normalized}, ensure_ascii=False, separators=(",", ":")))
